src/generate-grpc-web.ts

```diff
@@ -231,5 +250,154 @@ return new Promise((resolve, reject) => {
             grpc
           )
         )
+    )
+    .addFunction(
+      FunctionSpec.create('invoke')
+        .addTypeVariable(t)
+        .addParameter('methodDesc', t)
+        .addParameter('_request', TypeNames.ANY)
+        .addParameter('metadata', maybeMetadata)
+        .returns(TypeNames.anyType('Observable@rxjs').param(TypeNames.ANY))
+        .addCodeBlock(
+          CodeBlock.empty().add(
+            `const DEFAULT_TIMEOUT_TIME: number = 3 /* seconds */ * 1000 /* ms */;
+            const request = { ..._request, ...methodDesc.requestType };
+            const maybeCombinedMetadata =
+    metadata && this.options.metadata
+      ? new %T({ ...this.options?.metadata.headersMap, ...metadata?.headersMap })
+      : metadata || this.options.metadata;
+return new Observable(observer => {
+      const upStream = (() => {
+        %T.invoke(methodDesc, {
+          host: this.host,
+          request,
+          metadata: maybeCombinedMetadata,
+          transport: grpc.WebsocketTransport(),
+          debug: this.options.debug,
+          onMessage: (next) => {
+            observer.next(next as any);
+          },
+          onEnd: () => {
+            setTimeout(() => {
+              upStream();
```

Comment:
Curious about this, why do we recall `upStream` and so recall `grpc.invoke` when `onEnd` is called?

Reply:
The standard behavior for a grps stream is to always be active and close on the backend side, but since all http 1.1 proxies have a ttl for streams, we must reconnect automatically. (By ther way, we can put the possibility of reconnection in the arguments as an option, why not).

Reply:
Ah cool, that makes sense / I'm good with this as-is. Thanks!